pkg/oauthserver/authenticator/password/keystonepassword/keystonepassword.go

```diff
@@ -76,7 +146,14 @@ func (a keystonePasswordAuthenticator) AuthenticatePassword(username, password s
 		return nil, false, err
 	}
 
-	identity := authapi.NewDefaultUserIdentityInfo(a.providerName, username)
+	providerUserID := username
+	if a.useKeystoneIdentity {
+		providerUserID = userid
```

Comment:
only do the GetUserIDv3 lookup in this block... the data is discarded otherwise

Reply:
This function does two things - 1. authenticates users, 2. pulls out their keystone ids. You may notice that I changed openstack.AuthenticateV3 to GetUserIDv3, and they both have the same behavior, except the latter also returns an id.
Here is the code of AuthenticateV3 that we used before https://github.com/rackspace/gophercloud/blob/master/openstack/client.go#L149-L207